File: dashlane-project/Dashlane/src/main/java/com/dashlane/listeners/edittext/NoLockEditTextWatcher.java
package com.dashlane.listeners.edittext;

import android.text.Editable;
import android.text.TextWatcher;

import com.dashlane.dagger.singleton.SingletonProvider;
import com.dashlane.session.Session;


public class NoLockEditTextWatcher implements TextWatcher {
    @Override
    public void beforeTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void onTextChanged(CharSequence charSequence, int i, int i1, int i2) {

    }

    @Override
    public void afterTextChanged(Editable editable) {
        Session session = SingletonProvider.getSessionManager().getSession();
        if (session == null) return;
        SingletonProvider.getComponent().getLockRepository()
                         .getLockManager(session).setLastActionTimestampToNow();
    }
}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/listeners/edittext/NoLockEditTextWatcher.kt
package com.dashlane.listeners.edittext

import android.text.Editable
import android.text.TextWatcher
import com.dashlane.lock.LockManager

class NoLockEditTextWatcher(private val lockManager: LockManager) : TextWatcher {
    override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) = Unit
    override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) = Unit
    override fun afterTextChanged(editable: Editable) {
        lockManager.setLastActionTimestampToNow()
    }
}